r2-core/src/main/java/com/linkedin/r2/transport/http/client/SmoothRateLimiter.java

```diff
@@ -207,6 +250,13 @@ public void loop()
         return;
       }
 
+      // the size of the pending cannot be ever greater than the maxBuffered. We prefer
+      // running above the limit then risking a leak
```

Comment:
Just wondering if the queue is full, does it mean the enqueue rate and dequeue rate would be exactly the same at that point? Basically if the queue is full, we would not perform any rate limiting? Just want to get some clarification on the expected behavior.

Reply:
Exactly!